noir-projects/aztec-nr/aztec/src/state_vars/global_private_immutable.nr

```diff
@@ -0,0 +1,156 @@
+use crate::{
+    context::{PrivateContext, UtilityContext},
+    note::{
+        lifecycle::create_note,
+        note_getter::{get_note, view_note},
+        note_interface::{NoteHash, NoteType},
+        note_message::NoteMessage,
+    },
+    oracle::notes::check_nullifier_exists,
+    state_vars::state_variable::StateVariable,
+};
+
+use protocol_types::{
+    constants::GENERATOR_INDEX__INITIALIZATION_NULLIFIER, hash::poseidon2_hash_with_separator,
+    traits::Packable,
+};
+
+mod test;
+
+/// GlobalPrivateImmutable is a state variable that represents a contract-global private value that is set once and
+/// remains unchanged forever. Unlike [crate::state_vars::private_immutable::PrivateImmutable] which is "owned"
+/// (requiring wrapping in an [crate::state_vars::owned::Owned] state variable), GlobalPrivateImmutable is used directly
+/// in storage:
+///
+/// ```noir
+/// #[storage]
+/// struct Storage<Context> {
+///     your_variable: GlobalPrivateImmutable<YourNote, Context>,
+/// }
+/// ```
+///
+/// The key difference from PrivateImmutable is that GlobalPrivateImmutable has a predictable initialization nullifier
+/// that depends only on the storage slot (not on any owner's secret key). This makes it suitable for contract-global
+/// data because the state variable can be initialized only once per contract (and not once per contract-owner pair
+/// as is the case with `Owned<PrivateImmutable<YourNote, Context>, Context>`).
+///
+/// ## Example
+///
+/// A contract's configuration parameters can be represented as a GlobalPrivateImmutable. Once set during contract
+/// deployment or initial setup, these parameters remain constant for the lifetime of the contract. For example, an
+/// account contract's signing public key is typically stored using GlobalPrivateImmutable.
+///
```

Comment:
I'd add a section on Privacy mentioning that the fact the value was initialized is leaky because there is no nsk (as there is no owner).